resotocore/resotocore/config/config_override_service.py

```diff
@@ -0,0 +1,133 @@
+        for config_file in config_files:
+            with config_file.open() as f:
+                try:
+                    raw_yaml = yaml.safe_load(f)
+                    with_config_id = {config_file.stem: raw_yaml}
+                    merged = merge_json_elements(overrides_json, with_config_id)
+                    overrides_json = cast(Json, merged)
+                except Exception as e:
+                    log.warning(f"Can't read the config override {config_file}, skipping. Reason: {e}")
+
+        def is_dict(config_id: str, obj: Any) -> bool:
+            if not isinstance(obj, dict):
+                if not silent:
+                    log.warning(f"Config override with id {config_id} contains invalid data, skipping.")
+                return False
+            return True
+
+        # dict with all overrides for all config ids, such as resoto.core, resoto.worker, etc.
+        all_config_overrides: Dict[ConfigId, Json] = {
+            ConfigId(k): v for k, v in overrides_json.items() if is_dict(k, v)
+        }
+
+        if all_config_overrides is None:
+            if not silent:
+                log.warning("No config overrides found.")
+            return {}
+
+        return all_config_overrides
+
+    def get_override(self, config_id: ConfigId) -> Optional[Json]:
+        return self.overrides.get(config_id)
+
+    def get_all_overrides(self) -> Dict[ConfigId, Json]:
+        return self.overrides
+
+    def watch_for_changes(self) -> None:
+        async def watcher() -> None:
+            while not self.stop_watcher.is_set():
+                # all config files that needs to be checked for changes
+                config_files: List[Path] = []
+                # do a flatmap on directories
+                for path in self.override_paths:
+                    if await aos.path.isdir(path):
+                        config_files.extend(
+                            [
+                                Path(entry.path)
+                                for entry in await aos.scandir(path)  # scandir avoids extra syscalls
+                                if entry.is_file() and Path(entry.path).suffix in (".yml", ".yaml")
+                            ]
+                        )
+                    else:
+                        config_files.append(path)
+
+                # a quick optimization to avoid reading the files if none has been changed
+                mtime_hash = 0
+                config_files = sorted(config_files)
+                for file in config_files:
+                    mtime_hash = hash((mtime_hash, (await aos.stat(file)).st_mtime))
+
+                if mtime_hash == self.mtime_hash:
+                    await asyncio.sleep(self.sleep_time)
+                    continue
+                self.mtime_hash = mtime_hash
+
+                # fine to block here since it happens once in a lifetime
+                overrides = self._load_overrides(silent=True)
```

Comment:
consider making it run async
```suggestion
                overrides = await run_async(self._load_overrides, silent=True)
```